File: services/graph4j_service/src/process/src/main/java/process/pubsub/AlgorithmsDataPublisher.java
package process.pubsub;

import org.json.JSONObject;
import process.network.Envelope;

public class AlgorithmsDataPublisher extends Publisher {
    private final String fileId;
    private final String algorithmName;
    private final String algorithmType;
    private final JSONObject envelope;

    public AlgorithmsDataPublisher(String fileId, String algorithmName, String algorithmType) {
        super(System.getenv("DATA_AGGREGATOR_TOPIC"), System.getenv("ALGORITHMS_DATA_KEY"), "topic");
        this.fileId = fileId;
        this.algorithmName = algorithmName;
        this.algorithmType = algorithmType;
        this.envelope = Envelope.sendAlgorithmData(fileId, algorithmName, algorithmType);
    }

    public void send(JSONObject data) {
        envelope.put("payload", data);
        super.send(envelope);
    }
}

File: services/networkx_service/src/process/pubsub/algorithms_data_publisher.py
import os
import json
from .publisher import Publisher
from network.envelope import Envelope


class AlgorithmsDataPublisher(Publisher):
    def __init__(self, file_id, algorithm_name, algorithm_type):
        super().__init__(os.environ["DATA_AGGREGATOR_TOPIC"], os.environ['ALGORITHMS_DATA_KEY'], 'topic')
        self.file_id = file_id
        self.algorithm_name = algorithm_name
        self.algorithm_type = algorithm_type
        self.envelope = Envelope.send_algorithm_data(self.file_id, self.algorithm_name, self.algorithm_type)

    def send(self, data):
        self.envelope['payload'] = data

        return super().send(json.dumps(self.envelope))
